### ISSUE TITLE
feat(training): GPU-resident TensorDict replay buffer with n-step return + symmetry augmentation

### ISSUE THREAD

**Issue by cagataycali** [strands-labs/robots#861]
Part of #858 epic. Foundation for FastTD3 / FastSAC.

## Problem
No replay buffer in tree. FastTD3 and FastSAC both keep the **entire**
buffer on GPU (default `buffer_size = 1024 × num_envs`) — CPU↔GPU shuffle
is the bottleneck otherwise.

## Goal
`strands_robots.training.replay.SimpleReplayBuffer` matching Holosoma's
`SimpleReplayBuffer` API:

```python
buf = SimpleReplayBuffer(
    n_env=1024, buffer_size=1024, n_obs=D_obs, n_act=D_act,
    n_critic_obs=D_critic, n_steps=1, gamma=0.99, device="cuda",
)
buf.extend(transition: TensorDict)         # add one vec-env step
batch = buf.sample(batch_size)             # returns TensorDict with effective_n_steps
```

Required fields per transition:
- `observations`, `actions`, `critic_observations`
- `next.observations`, `next.critic_observations`, `next.rewards`,
  `next.dones`, `next.truncations`, `next.effective_n_steps`

## Scope
- n-step return with proper bootstrap masking (truncations vs dones — the
  bug FastTD3 fixed in Jun 2025).
- Optional symmetry augmentation hook (filed alongside in #C3 — symmetry
  utils land there; this issue exposes the augment-time integration point).
- AMP-friendly: pure tensor ops, no python-level shuffling per step.
- `to(device)`, `state_dict()`, `load_state_dict()` for checkpointing.

## Acceptance
- [ ] `SimpleReplayBuffer` in `strands_robots/training/replay.py`
- [ ] Unit tests for n=1 and n>1 with manual ground-truth on toy returns
- [ ] Bootstrap-mask test: truncation → bootstrap, true done → no bootstrap
- [ ] Survives 1M extend+sample cycles on a single A100 without leaks

## Refs
- Holosoma: `src/holosoma/holosoma/agents/fast_sac/fast_sac_utils.py`
- FastTD3: `fast_td3/fast_td3_utils.py` (uses TensorDict + Buffer)
- FastSAC standalone: `isaaclab_fast_sac/replay_buffer.py`
- n-step bug fix credit: @araffin, FastTD3 readme Jun 06 2025

**Comment by cagataycali** [comment]
Closing as out of scope for 1.0, and folding the tracking back into the parent epic #858.

The 0.5 -> 1.0 roadmap has six themes - tree-shaking, the dashboard, sealing the policy-provider list, a docs/examples rework, a starter kit and the Sprout embodiment. An off-policy RL training stack with parallel-sim backends (IsaacLab, MJWarp, MuJoCo Playground, HumanoidBench) is not among them, and it is a large enough body of work that keeping fifteen open children makes the 1.0 board unreadable.

Measured before closing, so this is not being written off as already-done: four of these children look shipped if you grade them by whether a class exists, and none of them meets its own acceptance criteria. `FastSacTrainer`, `PpoTrainer`, `VecSimEnv` and `elastic_launch_callable` are all present, but `num_atoms`/`c51`/`v_min` appear nowhere (so #859's distributional critic did not land), `adaptive_kl`/`desired_kl` appear nowhere (#865), `mjwarp` appears nowhere (so #863's vec-env does not span the three backends it names), and `elastic_launch_callable` has zero references from `training/rl/` (#870 asked for a launcher FOR the RL trainers). A symbol is a claim; the criteria are the contract.

#858 stays open as the single post-1.0 tracker for this stack. Re-file any individual piece against the 1.0+ roadmap when the RL lane is scheduled.

Closed as part of the 0.5 -> 1.0 backlog cleanup.
